Add verbose mode to Reconciler

ChrisBr opened the pull request:
to print some logs to STDOUT.

magnusvk commented on lib/counter_culture/reconciler.rb:
```diff
@@ -101,12 +102,23 @@ def perform
+        @log ||= if options[:verbose]
+                   Logger.new(STDOUT)
+                 else
+                   Logger.new('/dev/null')
```
I don't love this; some thoughts / questions:
1. Why not use `Rails.logger` here, at least if it's available?
2. I wouldn't use a logger to `/dev/null`, I think I'd prefer us to encapsulate this with a `log(message)` method that logs only if verbose is true.

What do you think?

ChrisBr commented:
@magnusvk thanks for your valuable feedback. I changed the code and also added a spec for it. Please have a look at it again and let me know what you think 👀

magnusvk commented:
@ChrisBr thanks! Going to release this as 2.2.0 when the specs go green.
